openpype/hosts/max/plugins/publish/extract_tycache.py

```diff
@@ -25,49 +26,50 @@ class ExtractTyCache(publish.Extractor):
         self.log.debug("Extracting Tycache...")
 
         stagingdir = self.staging_dir(instance)
-        filename = "{name}.tyc".format(**instance.data)
-        path = os.path.join(stagingdir, filename)
-        filenames = self.get_files(instance, start, end)
-        additional_attributes = instance.data.get("tyc_attrs", {})
 
-        with maintained_selection():
-            job_args = self.get_export_particles_job_args(
-                instance.data["members"],
-                start, end, path,
-                additional_attributes)
-            for job in job_args:
-                rt.Execute(job)
+        export_mode = instance.data.get("exportMode", 2)
+        members = instance.data["members"]
         representations = instance.data.setdefault("representations", [])
-        representation = {
-            'name': 'tyc',
-            'ext': 'tyc',
-            'files': filenames if len(filenames) > 1 else filenames[0],
-            "stagingDir": stagingdir,
-        }
-        representations.append(representation)
-
-        # Get the tyMesh filename for extraction
-        mesh_filename = f"{instance.name}__tyMesh.tyc"
-        mesh_repres = {
-            'name': 'tyMesh',
-            'ext': 'tyc',
-            'files': mesh_filename,
-            "stagingDir": stagingdir,
-            "outputName": '__tyMesh'
-        }
-        representations.append(mesh_repres)
-        self.log.debug(f"Extracted instance '{instance.name}' to: {filenames}")
-
-    def get_files(self, instance, start_frame, end_frame):
+        with maintained_selection():
+            for operators in get_operators(members):
+                operator, start_frame, end_frame, name = operators
+                filename = f"{instance.name}_{name}.tyc"
+                path = os.path.join(stagingdir, filename)
+                filenames = self.get_files(
+                    instance, name, start_frame, end_frame)
+                job_args = self.get_export_particles_job_args(
+                    operator, path, export_mode)
+                for job in job_args:
+                    rt.Execute(job)
+                representation = {
+                    'name': 'tyc',
+                    'ext': 'tyc',
+                    'files': filenames if len(filenames) > 1 else filenames[0],
+                    "stagingDir": stagingdir
+                }
+                representations.append(representation)
+
+                # Get the tyMesh filename for extraction
+                mesh_filename = f"{instance.name}_{name}__tyMesh.tyc"
+                mesh_repres = {
+                    'name': 'tyMesh',
+                    'ext': 'tyc',
+                    'files': mesh_filename,
+                    "stagingDir": stagingdir,
+                    "outputName": 'tyMesh'
+                }
+                representations.append(mesh_repres)
+                self.log.debug(
+                    f"Extracted instance '{instance.name}' to: {filenames}")
```

Comment:
This will create multiple representations with the same representation name if multiple "operators" are among the members which should never be valid. The names must then be unique.

Multiple representations with the same name sounds like the root problem of the issue described [here](https://github.com/ynput/OpenPype/pull/5930#pullrequestreview-1809457187).

Reply:
tycache would export tyPart(which happens to have frame range) and tyMesh respectively during extraction.